include/crocoddyl/core/squashing-base.hpp

```diff
@@ -0,0 +1,91 @@
+#ifndef CROCODDYL_CORE_SQUASHING_BASE_HPP_
+#define CROCODDYL_CORE_SQUASHING_BASE_HPP_
+
+#include <stdexcept>
+#include <boost/shared_ptr.hpp>
+#include <boost/make_shared.hpp>
+
+#include "crocoddyl/core/fwd.hpp"
+#include "crocoddyl/core/mathbase.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+
+// struct SquashingDataAbstractTpl; // forward declaration
+
+template <typename _Scalar>
+class SquashingModelAbstractTpl {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef SquashingDataAbstractTpl<Scalar> SquashingDataAbstract;
+  typedef typename MathBase::VectorXs VectorXs;
+
+  SquashingModelAbstractTpl(const std::size_t& ns) : ns_(ns) {
+    if (ns_ == 0) {
+      throw_pretty("Invalid argument: "
+                   << "ns cannot be zero");
+    }
+  };
+  virtual ~SquashingModelAbstractTpl(){};
+
+  virtual void calc(const boost::shared_ptr<SquashingDataAbstract>& data, const Eigen::Ref<const VectorXs>& u) = 0;
+  virtual void calcDiff(const boost::shared_ptr<SquashingDataAbstract>& data, const Eigen::Ref<const VectorXs>& u,
+                        const bool& recalc = true) = 0;
+  virtual boost::shared_ptr<SquashingDataAbstract> createData() {
+    return boost::make_shared<SquashingDataAbstract>(this);
+  }
+
+  const std::size_t& get_ns() const;
+  const VectorXs& get_in_lb() const { return in_lb_; };
```

Comment:
Let's rename this function as:
```cpp
const VectorXs& get_u_lb() const { return u_lb_; };
```
and apply the same rule for the other functions.

Reply:
In the paper we called the input to the squashing function "s".
I understand that we could call it "u" since it is our decision variables.

How should we call the output of the squashing, which is now indicated by "out"? Independently from the paper notation, we could use "s" (this is how it is defined in the SquashingData struct).

Reply:
I was confused indeed.

Then, I think we should call them as:
  - input: `s`
  - output: `u`

Do you agree with me?

Reply:
It makes more sense, yes. I will change how these variable are defined in the SquashingData struct too.